### ISSUE TITLE
angular-file-upload callback not work correctlly in firefox

### ISSUE THREAD

**Issue by darkmanAmazigh** [nervgh/angular-file-upload#351]
Hi,

I have an error when i upload file using angular-file-upload juste in firefox, in chrome it work well, 
all response is in uploader.onErrorItem instead uploader.onSuccessItem in firefox, 

nb:my code Server work fine,

Any one can help me,
thank you

**Comment by zackiles** [comment]
I can help you, use a different library. There are so many issues with this library, and the maintainer is not around. It's such a mess, just try running the src through any linter and you'll see what I mean. I've noticed what you said about Firefox, also ipad users on Safari it has problems with.

**Comment by demisx** [comment]
Agreed. Doesn't look like this package is getting enough care that it deserves.

**Comment by tspaeth** [comment]
:+1: unfortunately. So how about a maintained fork as the license will allow that?!

**Comment by andreaslarssen** [comment]
So, what lib did you guys end up using?

**Comment by nicolefinnie** [comment]
is ng-file-upload recommended? 
I currently hit a similar problem with angular-file-upload 1.1.5 and upgraded to 2.2.0 (the latest) and it doesn't solve the problem - the select file window doesn't pop up on Firefox but on Chrome, so I'm thinking of using another library instead.
